provider/internal/queue/queue.go

```diff
@@ -56,6 +56,12 @@ func (q *Queue) Close() error {
 
 // Enqueue puts a cid in the queue
 func (q *Queue) Enqueue(cid cid.Cid) error {
+	select {
+	case <-q.ctx.Done():
+		return fmt.Errorf("failed to enqueue CID: shutting down")
+	default:
+	}
+
```

Comment:
Why ? This is confusing and do nothing intresting.

Reply:
The initial function was non-deterministic and had a 50% chance to return an error when the context is done.

Reply:
But you probably know better the codebase so I trust you on this one

Reply:
@guillaumemichel we can't just write to `Enqueue` because it could deadlock, we still have to check the context.

My point is that `q.enqueue` will be backed up and the writes attempt to it would fail forcing select to go through the context.